coral/docs/development-with-local-klaw.md

```diff
@@ -0,0 +1,78 @@
+# Development with local Klaw
+
+ℹ️ The proxy is a work in progress.
+
+Please check out the [proxy README](../proxy/README.md) for more detailed information on how to use the proxy in your development process. You'll also find explanation and workaround for the things that are currently not working.
+
+## Table of content
+
+* [Basic setup](#basic-setup)
+* [First setup](#first-setup)
+* [How to run it](#how-to-run-it)
+
+ 
+## Basic setup
+
+ℹ️ environment settings are located in the file [`.env.local-api`](../../coral/.env.local-api).
+
+** ℹRequirements**
+
+- [node](https://nodejs.org/en/) needs to be installed <br/>
+  -> see [nvmrc](../.nvmrc) or the `engines` definition in [package.json](../package.json) for version).
+- Coral uses [pnpm](https://pnpm.io/) (version 7) as a package manager. Read their official documentation [how to install](https://pnpm.io/installation) pnpm.
+
+1. navigate to [`/coral`](../../coral)
+2. run `pnpm install`
+3. run `pnpm add-precommit` the first time you install the repository to set the custom directory for our pre commit hooks.
+- go to directory [`coral/proxy`](../../coral/proxy)
+- run `pnpm install` there, too
+4. Run development:
+   4.1. If you have not setup the docker environments, please follow [First setup]()
+   4.2. If you already have a setup, run `pnmv dev:[start|restart]`
+    - check out the [documentation](../../coral/proxy/README.md) for more scripts and information when to use them.
+    - the proxy runs on [`http://localhost:1337`](http://localhost:1337)
+  - ❗️ The correct redirect for login and authentication is **not** working yet. To authenticate yourself:
+    - Go to your [local Klaw](http://localhost:9097/login)
+    - Login with your credentials
+    - After you've been sucessfully logged in, go back to the proxy -> [`http://localhost:1337`](http://localhost:1337)
+
+🙋 The proxy and Klaw are running with `http`. Your browser may show you an error (SSL connection error), because it tries to automatically upgrade to https, without making an http request for security. You have to allow `localhost` to be accessed via http explicitly. 
+
+ℹ️ When you're done, you can run either: 
+- `pnpm dev:stop` to stop all containers in docker (enables a fast restart) 
+- `pnpm dev:destroy` to tear down all containers in docker (you'll have to run `pnpm setup` or `pnpm:start` again next time you want to use them, so they get build again)
+
+💡currently the proxy also does not redirect you to the login if your access expires. If you're getting related errors from your API, please login again like described 🙏
+
+
+## First setup
+
+1. Install Docker - [Get Docker](https://docs.docker.com/get-docker/)
+2. Go to to directory [`coral/proxy`](../../coral/proxy)
+3. run `pnpm install` if you haven't already
+4. Run `ppm setup`
+   `pnpm setup` will build and deploy your docker container for Klaw core, cluster api and a test environment for klaw (zookeeper, kafka, schema-registry).
```

Comment:
I think we may want to be more detailed about the role and responsability of each of these containers?

> `pnpm setup` will build and deploy docker containers for:
> - Klaw core: the main API Coral interacts with
> - Klaw cluster api: the API interacting with the Kafka clusters managed by Klaw
> - Kafka, zoo-keeper, schema-registry: a basic Kafka setup
>

Reply:
Yes, I can add that/ Have to see where :D because in the list it will probably not be a good place.

Reply:
I'm also thinking about if we need the sandbox setup for the `pnpm setup` 🤔  

The files are large, so maybe it's not a good idea to deploy all of them if users may not explicit need them. With this we're forcing users to do that (if they use `pnpm setup`). 

WDYT?

Reply:
I think if a user would wish to pick and choose the things done by `pnpm setup`, they are adept enough to not run `pnpm setup` and do it as they see fit ^^

As I see it, these are instructions for people coming into the project and just wanting the easiest way to set it up, even if it's not the most efficient.